### PR TITLE
dont attempt assignment to disposed ValueNotifiers

### PR DESCRIPTION
**Pull request by sevenrats**
dont attempt to assign to disposed ValueNotifiers.
This is especially abig problem in the search screen.

### CODE REVIEW

File: lib/components/paginated_item_carousel.dart [hunk 1]
```diff
@@ -51,13 +51,13 @@ class _FutureItemCarouselState extends ConsumerState<PaginatedItemCarousel> {
...
 
     if (data.isEmpty && !isLastPage.value) {
-      isLoading.value = true;
+      if (isLoading.hasListeners) {
+        isLoading.value = true;
+      }
       widget.future(startIndex, widget.pageSize).then((value) {
         data.addAll(value);
         startIndex += widget.pageSize;
-        isLoading.value = false;
```
Comment (jdk-21) [inline lib/components/paginated_item_carousel.dart:58]: `isLoading.value = false` must be present or else you're breaking the widget.

File: lib/components/paginated_item_carousel.dart [hunk 2]
```diff
@@ -51,13 +51,13 @@ class _FutureItemCarouselState extends ConsumerState<PaginatedItemCarousel> {
...
-
-        if (value.length < widget.pageSize) {
+        if (value.length < widget.pageSize && isLastPage.hasListeners) {
```
Comment (jdk-21) [inline lib/components/paginated_item_carousel.dart:60]: .hasListeners should be used with "normal" ChangeNotifiers not with a Hook.

**Review by jdk-21 — COMMENTED** [review]
SearchScreen isn't using PaginatedItemCarousel only FutureItemCarousel. Or maybe I'm not understanding the issue that this PR is supposed to solve.

**Comment by sevenrats** [comment]
then my guess is that this is the result of browing AWAY from homescreen before its done loading, or some interaction of that browsing away and searching.
since this app doesn't really need to use components that are not visible mostly, it would be nice to lock all the renderers somehow so that they can only run when the screen is shown.

File: lib/components/paginated_item_carousel.dart (hunk 2 above)
Comment (sevenrats) [reply lib/components/paginated_item_carousel.dart:63]: no clue what this means, sorry. do you mind pushing the relevant change yourself so I know what you're talking about?

Comment (jdk-21) [reply lib/components/paginated_item_carousel.dart:63]: calling a hook like isLoading.hasListeners produces a warning: The member 'hasListeners' can only be used within instance members of subclasses of 'package:flutter/src/foundation/change_notifier.dart'.

But you could use context.mounted instead. This checks if the current widget is mounted, which does essentially the same without warnings.

**Review by jdk-21 — APPROVED** [review]
Thank you for tracking down the issue.

**Pull request merged**